CMakeLists.txt

```diff
@@ -43,281 +44,260 @@ else()
 option(USE_OMP    "Build tiny-cnn with OMP library support" OFF)
 option(USE_NNPACK "Build tiny-cnn with NNPACK library support" OFF)
+option(USE_OPENCV "Build tiny-cnn with OpenCV library support" OFF)
 
 option(BUILD_TESTS "Set to ON to build tests" OFF)
-option(BUILD_EXAMPLES "Set to ON to build examples" ON)
+option(BUILD_EXAMPLES "Set to ON to build examples" OFF)
 option(BUILD_DOCS "Set to ON to build documentation" OFF)
 
-# ----------------------------------------------------------------------------
-#   Find Dependencies
-# ----------------------------------------------------------------------------
-
+# -------------- Dependencies --------------- #
 # Using cmake scripts and modules
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
 
+# Examples can be built using either with OpenCV or std_image.
 if(BUILD_EXAMPLES)
+    # Find Open Source Computer Vision (OpenCV)
     find_package(OpenCV)
     if(OpenCV_FOUND)
-      add_definitions(-DUSE_OPENCV)
-      include_directories(${OpenCV_INCLUDE_DIRS})
-      set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${OpenCV_LIBS})
-    endif()
-endif()
-
+        set(USE_OPENCV ON)
+        add_definitions(-DCNN_USE_OPENCV)
+        include_directories(${OpenCV_INCLUDE_DIRS})
+        set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${OpenCV_LIBS})
+    else(OpenCV_FOUND)
+        # OpenCV is not a real dependency. In case is not found,
+        # USE_OPENCV flag will be kept to OFF.
+        message(WARNING "Can't found OpenCV. Building examples without it.")
+    endif(OpenCV_FOUND)
+endif(BUILD_EXAMPLES)
+
+# Tiny-cnn provides a couple of multithreading solutions.
+# The user can specify to use Intel Threading Building Blocks (TBB)
+# or Open Multi-Processing (OpenMP) as a backend for multi threading
+# processing. In case that none of this libraries are required, tiny-cnn
+# will use the standard C++11 Thread support library.
+
+# Find Intel Threading Building Blocks (TBB)
 if(USE_TBB)
     find_package(TBB REQUIRED)
-    add_definitions(-DCNN_USE_TBB)
-    include_directories(${TBB_INCLUDE_DIRS})
-    link_directories(${TBB_LIBRARY_DIRS})
-    set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${TBB_LIBRARIES})
-elseif(USE_OMP)
+    if(TBB_FOUND)
+        message(STATUS "Found Intel TBB: ${TBB_INCLUDE_DIR}")
+        # In case that TBB is found we force to disable OpenMP since
+        # tiny-cnn does not support mutiple multithreading backends.
+        set(USE_OMP OFF)
+        add_definitions(-DCNN_USE_TBB)
+        include_directories(${TBB_INCLUDE_DIRS})
+        link_directories(${TBB_LIBRARY_DIRS})
+        set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${TBB_LIBRARIES})
+    else(TBB_FOUND)
+        # In case the user sets the flag USE_TBB to ON, the CMake build-tree
+        # will require to find TBB in your system. Otherwise, the user can
+        # set the paths to headers and libs by hand.
+        message(FATAL_ERROR "Can't find TBB. Please set TBB_INCLUDE_DIRS & "
```

Comment:
do you really want the user to set that by hand ? The user should modify his env variables by hand instead no ?